### ISSUE TITLE
Media notification disappears after a few seconds on some media players

### ISSUE THREAD

**Issue by jonalmeida** [mozilla-mobile/android-components#4116]
I've seen this happen a few times but I was able to figure out it was only for certain video players.

## STR
1. Play the video on the homepage of https://plyr.io/ (seems more prevalent with http://afterglowplayer.com/)
2. Observe the media notification in the system tray.
3. Wait for 15-20s

## Expected
1. The media notification is persistent through the entire. length of the video.

## Actual
1. The video continues to play on but the media notification disappears.

May require several attempts.

**Comment by pocmo** [comment]
After that time GeckoView reports a playback state of `SUSPENDED`. From the docs:
> Sent when loading of the media is suspended. This may happen either because the download has completed or because it has been paused for any other reason.

**Comment by pocmo** [comment]
I ended up doing what I apparently suggested in this comment from 5 months ago:
https://github.com/mozilla-mobile/android-components/blob/e341f9373665721944f388653f1b54f63c91a211/components/concept/engine/src/main/java/mozilla/components/concept/engine/media/Media.kt#L67-L70

I added a simplified `state` that doesn't expose all media events (somehow the name `playbackState` is lying, it is a stream of events not a state). This guarantees that the state keeps consistent if intermittent events happen that do not actually change paused/playing behavior.

I'll have to update and write some tests before I can open a PR.
